src/lazy/text/buffer.rs

```diff
@@ -1062,7 +1062,10 @@ impl<'top> TextBufferView<'top> {
     /// Matches and returns a boolean value.
     pub fn match_bool(self) -> IonParseResult<'top, bool> {
         terminated(
-            alt((value(true, tag("true")), value(false, tag("false")))),
+            alt((
+                value(true, complete_tag("true")),
+                value(false, complete_tag("false")),
+            )),
```

Comment:
🗺️ This is the logic change. The rest of the changes in this file are (a) separating out `mismatch` and `incomplete` in the tests cases, and (b) introducing a macro to eliminate most of the nested functions, boilerplate code, and repeated code in the test cases.